### PR TITLE
Add redux for chat room

### PR DESCRIPTION
**Pull request by ZijiaZhang**
- [ ] All features are tested
- [x] Version is bumped
- [x] Changelog is updated
- [x] Documentation is updated if needed

### CODE REVIEW

File: src/frontend/chat_room/Actions/index.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export enum ChatRoomActions{
+    RECEIVE_MESSAGE,
+    SEND_MESSAGE
+}
+
+export const send_message = (text: string) => {
```
Comment (xddenise) [inline src/frontend/chat_room/Actions/index.ts:6]: When do we want to use underscore and lower case in naming variables

File: src/frontend/chat_room/reducer/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {MessageStatus, SingleMessage} from "../components/ChatRoomBubbles";
+import {combineReducers} from "redux";
+import update from 'react-addons-update';
+import {ChatRoomActions} from "../Actions";
+
+export interface ChatRoomState {
+    messages: Array<SingleMessage>;
```
Comment (xddenise) [inline src/frontend/chat_room/reducer/index.ts:7]: I wonder if you need to keep track of which user is sending which message.

Comment (ZijiaZhang) [reply src/frontend/chat_room/reducer/index.ts:7]: We only care about `Received` and `Sent` to render the messages. I think I can add one more field `Sender`.

File: src/frontend/shared/globleFunctions.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import {IUserProps} from "./interfaces/IUserProps";
+
+export function getCurrentUser(): IUserProps{
+    // TODO
+    return {avatarPath: './images/photoP.png', name: 'Test User'}
+}
```
Comment (xddenise) [inline src/frontend/shared/globleFunctions.ts:6]: Add the extra line?

**Pull request merged**